Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/DisabilityType.hs

```diff
@@ -0,0 +1,50 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module Domain.Action.UI.DisabilityType
+  ( disabilityList,
+  )
+where
+
+import Data.List (groupBy)
+import qualified Domain.Types.DisabilityType as DTL
+import Kernel.Beam.Functions as B
+import Kernel.Prelude
+import qualified Kernel.Storage.Esqueleto as Esq
+import qualified Storage.Queries.DisabilityType as QD
+
+makeDisabilityListAPIEntity :: [DTL.DisabilityType] -> [DTL.DisabilityListAPIEntity]
+makeDisabilityListAPIEntity response = map convertGroup groupedEntities
+  where
+    groupedEntities = groupBy (\a b -> a.tag == b.tag) response
+    convertGroup :: [DTL.DisabilityType] -> DTL.DisabilityListAPIEntity
+    convertGroup [] = DTL.DisabilityListAPIEntity {tag = "", messages = []}
+    convertGroup group@(res : _) =
+      DTL.DisabilityListAPIEntity
+        { tag = res.tag,
+          messages = map convertResponse group
+        }
+    convertResponse :: DTL.DisabilityType -> DTL.Message
+    convertResponse res =
```

Comment:
you can use here like this also

convertResponse :: DTL.DisabilityType -> DTL.Message
convertResponse  DTL.DisabilityType {..} =  DTL.Message {..}